pkg/cmd/server/start/start_master.go

```diff
@@ -265,11 +265,24 @@ func (o MasterOptions) RunMaster() error {
 		return kerrors.NewInvalid("MasterConfig", o.ConfigFile, validationResults.Errors)
 	}
 
-	if err := StartMaster(masterConfig); err != nil {
-		return err
+	if !o.MasterArgs.StartControllers {
+		masterConfig.Controllers = configapi.ControllersDisabled
 	}
 
-	return nil
+	if !o.MasterArgs.StartAPI {
+		openshiftConfig, err := origin.BuildMasterConfig(*masterConfig)
+		if err != nil {
+			return err
+		}
+
+		_, kubeConfig, err := buildKubernetesMasterConfig(openshiftConfig)
+		if err != nil {
+			return err
+		}
+		return StartControllers(openshiftConfig, kubeConfig)
```

Comment:
Maybe in the future.  But now, it's not worth the larger refactor.  I want to wait until we decide what to do with the UI and the oauth server